### DISCUSSION TITLE
Sveltekit: How to use with regular hook functions e.g. database calls?

### DISCUSSION THREAD

**Discussion by xpluscal** [nextauthjs/next-auth#7187] [Help]
### Question 💬

The current documentation states the "SvelteKitAuth" as a replacement for the hooks handler function.

It would be great to still be able to use hooks e.g. for querying user information from the DB and passing it to locals.
Is that currently still possible and if so, how?

Thank you!

### How to reproduce ☕️

Use SvelteKitAuth handle.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by intrnl** [comment]
You should be able to do something like...
```ts
import { type Handle } from '@sveltejs/kit';

import { SvelteKitAuth } from '@auth/sveltekit';

import { authOptions } from '$lib/server/auth';

const AuthHandle = SvelteKitAuth(authOptions);

export const handle = (({ event, resolve }) => {
	const { url } = event;

	if (url.pathname.startsWith(authOptions.prefix + '/')) {
		return AuthHandle({ event, resolve });
	}

	return resolve(event);
}) satisfies Handle;

```

This assumes `prefix` is set in your auth options, but you can easily just change the check to `'/auth/'`

**Reply by intrnl** [reply]
Actually I just realized this isn't ideal, because `AuthHandle` also sets up `locals.getSession`, so doing this would just break that.

You *can* fix it by setting it up on your own, by adding something like this after the auth prefix check

```
event.locals.getSession ??= () => getSession(request, authOptions);
```

It would be better if it provides a way to not have to do this though, let's see.
